### PR TITLE
MOSIP-28008: Added support in clientsettings sync for 1.1.4 regclient

### PR DESCRIPTION
**Pull request by HimajaDhanyamraju2**

### CODE REVIEW

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/entity/Gender.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+/**
+ * Gender entity mapped according to DB
+ * 
+ * @author Urvil Joshi
+ * @author Sidhant Agarwal
+ * @since 1.0.0
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/entity/Gender.java:24]: Kindly add a note here that this is deprecated, and its added only for backward compatibility

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/entity/IndividualType.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import io.mosip.kernel.syncdata.entity.id.CodeAndLanguageCodeID;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+/**
+ * 
+ * @author Srinivasan
+ *
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/entity/IndividualType.java:21]: Kindly add a note here that this is deprecated, and its added only for backward compatibility

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/repository/GenderRepository.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import io.mosip.kernel.syncdata.entity.Gender;
+
+/**
+ * Repository class for fetching gender data
+ * 
+ * @author Sidhant Agarwal
+ * @since 1.0.0
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/repository/GenderRepository.java:18]: Kindly add a note here that this is deprecated, and its added only for backward compatibility

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/repository/IndividualTypeRepository.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.stereotype.Repository;
+
+import io.mosip.kernel.syncdata.entity.IndividualType;
+import io.mosip.kernel.syncdata.entity.id.CodeAndLanguageCodeID;
+
+/**
+ * Repository class to handle CRUD operations for
+ * {@link IndividualTypeRepository}}
+ * 
+ * @author Srinivasan
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/repository/IndividualTypeRepository.java:19]: Kindly add a note here that this is deprecated, and its added only for backward compatibility

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/utils/SyncMasterDataServiceHelper.java [hunk 5]
```diff
@@ -1652,6 +1926,12 @@ private boolean isChangesFound(String entityName, LocalDateTime lastUpdated) {
 			case "DocumentCategory":
 				result = documentCategoryRepository.getMaxCreatedDateTimeMaxUpdatedDateTime();
 				break;
+			case "Gender":
+				result = genderRepository.getMaxCreatedDateTimeMaxUpdatedDateTime();
+				break;
+			case "IndividualType":
+				result = individualTypeRepository.getMaxCreatedDateTimeMaxUpdatedDateTime();
+				break;
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/utils/SyncMasterDataServiceHelper.java:1934]: I don't see Device, DeviceSpecification and DeviceType in this switch

**Review by ase-101 — CHANGES_REQUESTED** [review]
Also @HimajaDhanyamraju2  can you correct the URL [here](https://github.com/mosip/admin-services/blob/release-1.2.0.1/admin/kernel-syncdata-service/src/main/resources/application-dev.properties#L51) to point to masterdata service

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/entity/Gender.java (hunk 1 above)
Comment (HimajaDhanyamraju2) [reply admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/entity/Gender.java:24]: Added.

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/entity/IndividualType.java (hunk 2 above)
Comment (HimajaDhanyamraju2) [reply admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/entity/IndividualType.java:21]: Added

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/repository/GenderRepository.java (hunk 3 above)
Comment (HimajaDhanyamraju2) [reply admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/repository/GenderRepository.java:18]: Added

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/repository/IndividualTypeRepository.java (hunk 4 above)
Comment (HimajaDhanyamraju2) [reply admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/repository/IndividualTypeRepository.java:19]: Added

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/utils/SyncMasterDataServiceHelper.java (hunk 5 above)
Comment (HimajaDhanyamraju2) [reply admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/utils/SyncMasterDataServiceHelper.java:1944]: Added device related cases.

File: admin/kernel-syncdata-service/src/main/resources/application-local.properties [hunk 6]
```diff
@@ -177,4 +177,7 @@ syncdata.cache.evict.delta-sync.cron=0 0/15 * * * *
 syncdata.cache.snapshot.cron=0 0 23 * * *
 
 mosip.kernel.auth.appids.realm.map={prereg:'preregistration',ida:'mosip',registrationclient:'mosip',regproc:'mosip',partner:'mosip',resident:'mosip'}
-mosip.kernel.keymanager.unique.identifier.autoupdate=false
\ No newline at end of file
+mosip.kernel.keymanager.unique.identifier.autoupdate=false
+auth.server.admin.issuer.internal.uri=https://iam.dev.mosip.net/auth/realms/
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/resources/application-local.properties:181]: can you add these in the application-dev.properties as well

Comment (HimajaDhanyamraju2) [reply admin/kernel-syncdata-service/src/main/resources/application-local.properties:181]: Done.

**Review by ase-101 — APPROVED** [review]

**Pull request merged**